src/streamlit_extras/image_selector/__init__.py

```diff
@@ -0,0 +1,182 @@
+from __future__ import annotations
+
+from io import BytesIO
+from typing import Literal
+
+import numpy as np
+import plotly.graph_objects as go
+import requests
+import streamlit as st
+from PIL import Image, ImageDraw
+
+from streamlit_extras import extra
+
+
+def convert_to_pil_image(image: str | np.ndarray | Image.Image) -> Image.Image:
+    """
+    Converts an image from various sources (URL, local path, numpy array, or PIL.Image) to a PIL.Image object.
+
+    Parameters:
+    - image: Union[str, np.ndarray, Image.Image]
+        The input image which can be:
+        - URL (str) pointing to the image
+        - Local file path (str)
+        - Numpy array (np.ndarray)
+        - PIL.Image.Image object
+
+    Returns:
+    - Image.Image: The converted PIL.Image object.
+
+    Raises:
+    - ValueError: If the input type is not supported or the image cannot be opened.
+    """
+    if isinstance(image, str):
+        if image.startswith("http://") or image.startswith("https://"):
+            response = requests.get(image)
+            if response.status_code == 200:
+                image = Image.open(BytesIO(response.content))
+            else:
+                raise ValueError("Could not retrieve image from URL.")
+        else:
+            image = Image.open(image)
+    elif isinstance(image, np.ndarray):
+        image = Image.fromarray(image)
+    elif isinstance(image, Image.Image):
+        # Image is already a PIL.Image
+        pass
+    else:
+        raise ValueError("Unsupported image type.")
+
+    return image
+
+
+@extra
+def image_selector(
+    image: Image.Image | str | np.ndarray,
+    selection_type: Literal["lasso", "box"],
+    width: int = 300,
+    height: int = 300,
+) -> dict:
+    """Show the image, and enable the user to select an area in
+    the image using the provided selection type.
+
+    Args:
+        image (Image.Image | str | np.ndarray): Original image. Can be a PIL object,
+            or path to local file, or URL, or NumPy array
+        selection_type (Literal[["lasso", "box"]): Selection type
+        width (int, optional): Width of the image container. Defaults to 300.
+        height (int, optional): Height of the image container. Defaults to 300.
+
+    Returns:
+        dict: Selection coordinates
+    """
+
+    image = convert_to_pil_image(image)
```

Comment:
**nit**: My preference is to avoid overloading a variable name if it changes type